### ISSUE TITLE
[sql] Select the sql column datatype Decimal is not support.

### ISSUE THREAD

**Issue by XPH0816** [tauri-apps/plugins-workspace#1444]
I request as
`select price from room`

Here's my db table:
![image](https://github.com/tauri-apps/plugins-workspace/assets/50739818/0f32470d-4713-4827-a377-dc85c324ddde)

Result:
![image](https://github.com/tauri-apps/plugins-workspace/assets/50739818/62317775-6056-4332-b53a-ab1002f51799)

**Comment by FabianLars** [comment]
Thanks for the report! Are we talking about sqlite, mysql or postgresql?

**Comment by XPH0816** [comment]
for my situation, I just using mysql.

**Comment by FabianLars** [comment]
Ah i think i remember. Since DECIMAL can't (or shouldn't) be converted to f64 i wasn't sure how to best send it to the js side so i just left it out for later. Still don't know any better (i never use sql myself) so this probably has to wait for the plugin's partial rewrite.

**Comment by XPH0816** [comment]
Noted with Thanks

**Comment by Tenodec** [comment]
change the type by sql
`SELECT CAST(id AS SIGNED) as id FROM table`

**Comment by Tenodec** [comment]
change the type by sql.
for example:
`SELECT CAST(id AS SIGNED) as id FROM table`

**Comment by daniellcc** [comment]
To follow up on what @Tenodec said, it worked for me casting the decimal value as **CHAR**, otherwise it rounds the value in certain scenarios.
For example:
`SELECT CAST(price AS CHAR) as price FROM table`

**Comment by orpos** [comment]
I also got this error, but in postgresql.
There are some other types that are unsupported like arrays, real, decimal

Will the plugin partial rewrite add support for more types both in postgresql and in mysql?

**Comment by FabianLars** [comment]
> Will the plugin partial rewrite add support for more types both in postgresql and in mysql?

Not necessarily, no. This is not the goal of the rewrite and type support can be added before/after the rewrite (this part of the code shouldn't change much in the rewrite).
A large part of the rewrite already happened but i forgot about this issue.

**Comment by mikenikles** [comment]
I had the same issue, among a few other missing mysql types and fixed it with the following code in `tauri-plugin-sql/src/decode/mysql.rs` in my fork of the sql plugin:

```rust
"DECIMAL" => {
    if let Ok(v) = ValueRef::to_owned(&v).try_decode::<rust_decimal::Decimal>() {
        JsonValue::String(v.to_string())
    } else {
        JsonValue::Null
    }
}
```

I'll open a PR with that and the other missing types (`SET`, `GEOMETRY`, `BIT`, `BINARY`, `VARBINARY`) shortly.

**Comment by AbdenourTadjer33** [comment]
I have created a PR that fixes this issue #3286.
